### DISCUSSION TITLE
feature request: vim type file editing in powershell

### DISCUSSION THREAD

**Discussion by mieliespoor** [PowerShell/PowerShell#14801] [General]
It would be great if PowerShell could support something like a vim experience of editing files, but without needing to install other tools like vim. 

In scenarios where machines are headless offering no UI experience and not having the ability to install other tools, it becomes difficult to edit any file. It's easy enough to view the file content as you can simple do a Get-Content -Path {path}, but to then edit line 50 in a long file can become an effort that can easily be saved if a tool can be built into powershell.

This topic might lead to some wildly differing opinions, hence posting it here instead of under issues.

**Comment by jantari** [comment]
My 2ct:

If a machine is headless, you should probably also deploy to it headless-ly. So either an immutable image, or configure the state through a tool like DSC, ansible or puppet.

If you find you must manually edit a text file on a remote machine, you can also just install vim or neovim.

The usecase you are describing is very niche, frankly also because it is generally not considered a good practice to manually edit files on a remote server without first checking your changes into version control.

My tip would be to look into ansible, perhaps the `win_lineinfile` module, instead.

**Comment by kilasuit** [comment]
Personally I'd prefer this wasn't added in as there's better UX experience in using tooling like VSCode's remote development feature connect to a machine that is headless from your own local machine instead via SSH connection & edit files in actual editors - https://code.visualstudio.com/docs/remote/ssh-tutorial
If you are still connecting to older Windows systems you can use PowerShell ISE's remote connection too -https://devblogs.microsoft.com/powershell/powershell-remote-script-debugging-in-the-ise/
